File: swift/sprint5/J/solution.swift
#if !REMOTE_JUDGE
open class Node {

    public var value: Int
    public var left: Node?
    public var right: Node?

    public init(_ value: Int) {
        self.value = value
        self.right = nil
        self.left = nil
    }
}
#endif

func insert(root: Node, newValue: Int) -> Node {
    // Your code
    // ヽ(´▽`)/
    return root
}

func test() {
    var node1 = Node(7)
    var node2 = Node(8)
    node2.left = node1
    var node3 = Node(7)
    node3.right = node2
    var newHead = insert(root: node3, newValue: 6)
    assert(newHead === node3)
    assert(newHead.left!.value == 6)
}

File: csharp/sprint5/J/Solution.cs
#if !REMOTE_JUDGE
public class Node
{
    public int Value { get; set; }
    public Node Left { get; set; }
    public Node Right { get; set; }

    public Node(int value)
    {
        Value = value;
        Left = null;
        Right = null;
    }
}
#endif

public class Solution
{
    public static Node Insert(Node root, int key)
    {
        // Your code
        // “ヽ(´▽｀)ノ”
    }

    #if !REMOTE_JUDGE
    private static void Main()
    {
        var node1 = new Node(7);
        var node2 = new Node(8)
        {
            Left = node1
        };

        var node3 = new Node(7)
        {
            Right = node2
        };
        var newHead = Insert(node3, 6);
        Console.WriteLine(newHead == node3);
        Console.WriteLine(newHead.Left.Value == 6);
    }
    #endif
}
